setup.py

```diff
@@ -442,19 +442,18 @@ def run(self):
     if debugging_symbols_requested:
         extra_compile_args.append('-g')
 
-# For mac, ensure extensions are built for macos 10.9 when compiling on a
-# 10.9 system or above, overriding distuitls behaviour which is to target
-# the version that python was built for. This may be overridden by setting
-# MACOSX_DEPLOYMENT_TARGET before calling setup.py
+# For mac, ensure extensions are built for at least macOS 10.9 when compiling
+# on a 10.9 system or above, overriding CPython distuitls behaviour which is
+# to target the version that python was built for. This may be overridden by
+# setting MACOSX_DEPLOYMENT_TARGET before calling setup.py
 if is_platform_mac():
     if 'MACOSX_DEPLOYMENT_TARGET' not in os.environ:
         current_system = LooseVersion(platform.mac_ver()[0])
         python_target = LooseVersion(
-            get_config_var('MACOSX_DEPLOYMENT_TARGET'))
+            get_config_vars().get('MACOSX_DEPLOYMENT_TARGET', current_system))
```

Comment:
This line has to be
```
get_config_vars().get('MACOSX_DEPLOYMENT_TARGET', str(current_system)))
```
i.e. `LooseVersion` expects a string while `current_system` is already an `LooseVersion` object.

With this fix `pip install` passes on pypy3.6-7.1.1.

Reply:
thanks, I didnt realise `LooseVersion` cant accept it own type. I've fixed it in a slightly different way now.